src/Index/MappingBuilder.php

```diff
@@ -35,6 +47,8 @@ public function buildIndexMapping(IndexConfigInterface $indexConfig)
             $mappingIndex['settings'] = $settings;
         }
 
+        $this->dispatcher->dispatch(new PostMappingBuildEvent($indexConfig, $mapping));
```

Comment:
Just wondering, wouldn't it be more logical to dispatch the event just before the mapping is added to `$mappingIndex`?

With the current setup, if `$mapping` is empty, it would not be added to `$mappingIndex`, even if it gets altered by an event listener.

```php
        $mapping = $this->buildMapping($indexConfig->getModel(), $indexConfig);

        $this->dispatcher->dispatch(new PostMappingBuildEvent($indexConfig, $mapping));

        $mappingIndex = [];
        if (!empty($mapping)) {
            $mappingIndex['mappings'] = $mapping;
        }

        $settings = $indexConfig->getSettings();
        if (!empty($settings)) {
            $mappingIndex['settings'] = $settings;
        }
```

Reply:
Yes! https://github.com/FriendsOfSymfony/FOSElasticaBundle/pull/1734/commits/8fc39440df90cb8fb12a95f582b290a1e30636dd